src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -323,40 +308,26 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
     /**
      *  Given a FedoraResource, return a list of agents.
      */
-    private Set<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
+    private Stream<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
             final FedoraResource resource) {
-        final Set<String> members = new HashSet<>();
-        final Model model = createDefaultModel();
-
-        final Predicate<Property> isMember = memberTestFromTypes.apply(resource.getTypes());
-
-        resource.getTriples(translator, PROPERTIES)
-            .filter(p -> isMember.test(model.asStatement(p).getPredicate()))
-            .forEach(t -> {
-                if (t.getObject().isURI()) {
-                    members.add(t.getObject().getURI());
-                } else if (t.getObject().isLiteral()) {
-                    members.add(t.getObject().getLiteralValue().toString());
-                }
-            });
-
-        return members;
+        return resource.getTriples(translator, PROPERTIES).filter(memberTestFromTypes.apply(resource.getTypes()))
+            .map(Triple::getObject)
+            .map(object -> object.isURI() ? object.getURI() : object.getLiteralValue().toString());
```

Comment:
Maybe break this function on triple objects out into a private method, use it here as a method ref, and below (~363ish) as a method?